sympy/solvers/bivariate.py

```diff
@@ -199,17 +244,61 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in list(set(flatten(power_list(lhs, symbol)))) \
+                    if i%2 == 0 and i != 0]
+    t = Dummy('t', **symbol.assumptions0)
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    def _solve_even_degree_branches(diff, symbol, t, gens):
+        """
+        Helper function to consider even degree branches of lambert
+        equation if exists.
+
+        For example:
+        Equation (a/x + exp(x/2)).diff(x) = 0 is represented
+        in logform as both of the below equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Input: diff, symbol, t, gens
+        diff - Includes a dummy variable t which can be replaced
+        with +symbol and -symbol.
+
+        Output: Returns combined solution after considering all
+        even degree branches of original equation.
+        """
+        llhs1, llhs2 = map(lambda i: diff.xreplace({t: i}), (symbol, -symbol))
+        if llhs1 == llhs2:
+            return _solve_lambert(llhs1, symbol, gens)
+        else:
+            sol1, sol2 = map(lambda i: _solve_lambert(i, symbol, gens), (llhs1, llhs2))
+            return list(set(sol1 + sol2))
+
+    if lhs.is_Add and len(even_degrees) == 1:
+        lhs = lhs.xreplace({symbol**even_degrees[0]: t**even_degrees[0]})
+        if lhs.has(t):
```

Comment:
Is this check really needed? I mean `symbol` will be replaced by `t` everytime (by the previous line) if it gets in here.

Reply:
Yes here is an example
```python
>>> p = symbols('p', positive=True)
>>> eq = 4*2**(2*p + 3) - 2*p - 3
>>> _solve_lambert(eq, p, _filtered_gens(Poly(eq), p))
[-3/2 - LambertW(-4*log(2))/(2*log(2))]
>>> lhs = 4*2**(2*p + 3) - 2*p
>>> _power_list(lhs, p)
[(1,), (2,)]
# even_degrees = [2]
```
Clearly not any of symbol is being replaced.

Reply:
I think _power_list has to be changed because it is returning wrong answer here.

Reply:
Indeed, also I don't follow your explanation of why is the check ` if lhs.has(t):` is required.

Reply:
Also what would happen in the else clause?